azure/services/virtualnetworks/virtualnetworks.go

```diff
@@ -84,26 +89,30 @@ func (s *Service) Reconcile(ctx context.Context) error {
 
 	result, err := s.CreateOrUpdateResource(ctx, vnetSpec, serviceName)
 	if err == nil && result != nil {
-		existingVnet, ok := result.(network.VirtualNetwork)
+		existingVnet, ok := result.(armnetwork.VirtualNetwork)
 		if !ok {
-			return errors.Errorf("%T is not a network.VirtualNetwork", result)
+			return errors.Errorf("%T is not an armnetwork.VirtualNetwork", result)
 		}
 		vnet := s.Scope.Vnet()
 		vnet.ID = ptr.Deref(existingVnet.ID, "")
 		vnet.Tags = converters.MapToTags(existingVnet.Tags)
 
 		var prefixes []string
-		if existingVnet.VirtualNetworkPropertiesFormat != nil && existingVnet.VirtualNetworkPropertiesFormat.AddressSpace != nil {
-			prefixes = azure.StringSlice(existingVnet.VirtualNetworkPropertiesFormat.AddressSpace.AddressPrefixes)
+		if existingVnet.Properties != nil && existingVnet.Properties.AddressSpace != nil {
+			for _, prefix := range existingVnet.Properties.AddressSpace.AddressPrefixes {
+				if ptr.Deref(prefix, "") != "" {
```

Comment:
(same as above in `converters.GetSubnetAddressesV2`)

Reply:
Thank you so much for reviewing @nojnhuh! I think I've addressed all your comments.